scripts/analyze_data.py

```diff
@@ -0,0 +1,179 @@
+            bar.finish()
+
+            # Store csv with computed metrics
+            if args.create_csv:
+                # Based on https://stackoverflow.com/questions/8685809/writing-a-dictionary-to-a-csv-file-with-one-line-for-every-key-value
+                out_csv_folder = os.path.join(output_folder, 'files')
+                if not os.path.exists(out_csv_folder):
+                    os.makedirs(out_csv_folder)
+                csv_path_sum = os.path.join(out_csv_folder, f'computed_metrics_{key}.csv')
+                with open(csv_path_sum, 'w') as csv_file:  
+                    writer = csv.writer(csv_file)
+                    for metric_name, metric in sorted(metrics_dict[key].items()):
+                        if isinstance(metric,dict):
+                            for metric_value, count in sorted(metric.items()):
+                                k = f'{metric_name}_{metric_value}'
+                                writer.writerow([k, count])
+                        else:
+                            writer.writerow([metric_name, metric])
+                
+                # Based on https://github.com/spinalcordtoolbox/disc-labeling-benchmark 
+                csv_path_fprint = os.path.join(out_csv_folder, f'fprint_{key}.csv')
+                sub_list = [sub for sub in fprint_dict[key].keys() if sub.startswith('sub')]
+                fields = ['subject'] + [k for k in fprint_dict[key][sub_list[0]].keys()]
+                with open(csv_path_fprint, 'w') as f:  
+                    w = csv.DictWriter(f, fields)
+                    w.writeheader()
+                    for k, v in fprint_dict[key].items():
+                        w.writerow(mergedict({'subject': k},v))
+
+        
+        if missing_data:
+            print("missing files:\n" + '\n'.join(missing_data))
+
+    # Plot data informations            
+    save_graphs(output_folder=output_folder, metrics_dict=metrics_dict, data_form=data_form)
+
+
+
+                
+
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Analyse config file')
+    
+    ## Parameters
+    parser.add_argument('--paths-to-bids', default='', nargs='+',
+                        help='Paths to BIDS compliant datasets (You can add multiple paths using spaces)')
+    parser.add_argument('--config', default='',
+                        help='Path to JSON config file that contains all the training splits')
+    parser.add_argument('--paths-to-csv', default='', nargs='+',
+                        help='Paths to csv files with already computed metrics (You can add multiple paths using spaces)')
+    parser.add_argument('--split', default='ALL', choices=('TRAINING', 'VALIDATION', 'TESTING', 'ALL'),
+                        help='Split of the data that will be analysed (default="ALL")')
+    parser.add_argument('--create-csv', default=True,
+                        help='Store computed metrics using a csv file in results/files (default=True)')
```

Comment:
I believe it would be more "pythonic" to include these lines under a separate function, for example, `get_parser`, example [here](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/69a4d2f1e62c44016f34ada7c1593447b3c1e971/spinalcordtoolbox/scripts/sct_analyze_lesion.py#L24C5-L24C15) and [here](https://github.com/spinalcordtoolbox/manual-correction/blob/19284b4343f4e31e100f72e84bc8af36feff24b3/manual_correction.py#L39).